### DISCUSSION TITLE
Different Type of Users

### DISCUSSION THREAD

**Discussion by Esorethewarrior0623** [nhost/nhost#1043] [Q&A]
Hello, I am trying to create an auth system similar to that of something like Doordash, where you have different types of users like, the "business", "drivers" and "customers" that have different app interfaces, but still connect to the same database. Would this just be a change in the role like an enum of type : "business", "customer", "driver"? OR would there be much more separation on that. Any advice on how to approach this would be great

**Comment by elitan** [comment]
It depends very much on your use case.

I would recommend reading this from Hasura: https://hasura.io/docs/latest/auth/authorization/roles-variables/

What have you tried so far, and where have you found issues with your solution(s)?
